sds_data_manager/lambda_code/SDSCode/queries.py

```diff
@@ -28,14 +33,62 @@ def lambda_handler(event, context):
     logger.info(f"Context: {context}")
 
     logger.info("Received event: " + json.dumps(event, indent=2))
-    # TODO: Replace result using RDS
-    search_result = ""
-    logger.info("Query Search Results: " + json.dumps(search_result))
+
+    # add session, pick model like in indexer and add query to filter_as
+    query_params = event["queryStringParameters"]
+
+    # select the file catalog for the query
+    query = select(models.FileCatalog.__table__)
+    # get a list of all valid search parameters
+    valid_parameters = [
+        column.key
+        for column in models.FileCatalog.__table__.columns
+        if column.key not in ["id", "status_tracking_id"]
+    ]
+    # go through each query parameter to set up sqlalchemy query conditions
+    for param, value in query_params.items():
+        # confirm that the search parameter is valid
+        if param not in valid_parameters:
+            response = {
+                "statusCode": 400,
+                "body": json.dumps(
+                    f"{param} is not a valid search parameter. "
+                    + f"Valid search parameters are: {valid_parameters}"
+                ),
+                "headers": {
+                    "Content-Type": "application/json",
+                    "Access-Control-Allow-Origin": "*",  # Allow CORS
+                },
+            }
+            return response
+        # check if we're search for start_date or end date to
+        # setup the correct "where" time condition
+        if param == "start_date":
+            query = query.where(
+                models.FileCatalog.start_date
+                >= datetime.datetime.strptime(value, "%Y%m%d")
+            )
+        elif param == "end_date":
+            # TODO: Need to discuss as a team how to handle date queries. For now,
+            # the date queries will only look at the file start_date.
+            query = query.where(
+                models.FileCatalog.start_date
+                <= datetime.datetime.strptime(value, "%Y%m%d")
+            )
+        # all non-time string matching parameters
+        else:
+            query = query.where(getattr(models.FileCatalog, param) == value)
+
+    engine = db.get_engine()
+    with Session(engine) as session:
+        search_result = session.execute(query).all()
+
+    logger.info("Query Search Results: " + str(search_result))
```

Comment:
```suggestion
    logger.info("Found [%s] Query Search Results: %s", len(search_result), search_result)
```
Would it be helpful to have the number of results found for our logs too?